src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -187,8 +211,22 @@ private void AllocateBuffer()
                 }
                 else
                 {
+                    // Example scenario: Let's say _count == 256.
+                    // Then our buffers look like this: | 32 | 32 | 64 | 128 |
+                    // As you can see, our count will be just double the last buffer.
+                    // Now, say _maxCapacity is 500. We will find the right amount to allocate by
+                    // doing min(256 * 2, 500) - 256. This represents the total we've allocated after
+                    // this operation, minus the amount we've already allocated.
+
+                    // Note: In case _count overflows (last buffer is 0x40000000) or _count * 2
+                    // overflows (last buffer is 0x20000000), we must cast to uint for correctness
+                    // during comparisons.
+
+                    Debug.Assert((uint)_count >= (uint)ResizeLimit * 2);
```

Comment:
I don't understand why these casts are necessary.  We know that ResizeLimit is an int... it's a const int defined earlier, and it's defined to be positive.  We also know from the earlier assert that _count < _maxCapacity, and _maxCapacity has already been asserted to be greater than 0.  So if _count is negative, an assert would have already failed.